python/ray/tune/tune.py

```diff
@@ -252,6 +252,8 @@ def run(
     checkpoint_score_attr: Optional[str] = None,
     checkpoint_freq: int = 0,
     checkpoint_at_end: bool = False,
+    checkpoint_keep_all_ranks: bool = False,
+    checkpoint_upload_from_workers: bool = False,
```

Comment:
Instead of adding more attributes here, can we deprecate the old checkpoint_ attributes and accept a `CheckpointConfig`?

Reply:
let's do this together??
no need to complicate this PR.

Reply:
Fine with me, but let's slot this in before 2.5 then. Otherwise they become part of the public API... :-)

Reply:
sure, how do you want to do this? we need to add an additional CheckpointConfig parameter, then show warning messages if any of the checkpoint_ fields are specified, right?